lib/xdrgen/generators/rust/src/types.rs

```diff
@@ -1219,8 +1222,15 @@ impl<const MAX: u32> ReadXdr for VecM<u8, MAX> {
                 return Err(Error::LengthExceedsMax);
             }
 
-            let mut vec = vec![0u8; len as usize];
-            r.read_exact(&mut vec)?;
+            let mut vec = vec![0u8; 0];
+            let mut len_remaining = len as usize;
+            while len_remaining > 0 {
+                let len_read = core::cmp::min(len_remaining, MAX_PREALLOCATED_BYTES_READ);
+                let offset = vec.len();
+                vec.resize(vec.len() + len_read, 0);
+                r.read_exact(&mut vec[offset..])?;
+                len_remaining -= len_read;
```

Comment:
the only thing that I'm slightly concerned with here is just underflow paranoia around "what if the arithmetic is wrong or read_exact misbehaves"; maybe .saturating_sub() or .checked_sub()?

Reply:
Ah yes this is a really good observation, because read_exact is part of the trait and overridable rather than being a fixed library function. I'll change the logic.

Reply:
Looked into this more, the read_exact function can't communicate about how much it has read. It returns `()` in the success case. So the implementor of read_exact has no control over the `len_read` and `len_remaining` values, so I don't think any underflow risk exists.

I mean, the read_exact can misbehave, the data read might be corrupt, but the arithmetic won't meaningful play a role in that.

Reply:
yeah, I mean I see that it's as correct as my brain can convince itself of; I guess I am just trying to be extra paranoid and defensive anywhere "wrong arithmetic could be fatal" and here it'd be an infinite-read loop.but of course there are limits to healthy paranoia and if you think it's overkill I guess I won't block this.

Reply:
Yup, but read_exact can't affect the values of len_read and len_remaining. The only thing affecting them is the code in that function, so if there's a bug there, we are 100% responsible for it, not he implementor of read_exact.